File: apps/backend-java/src/main/java/com/housingplatform/features/notifications/dto/NotificationDto.java
package com.housingplatform.features.notifications.dto;

import com.housingplatform.persistence.enums.NotificationType;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

public record NotificationDto(
    UUID id,
    UUID userId,
    NotificationType type,
    String title,
    String body,
    Map<String, Object> metadata,
    OffsetDateTime readAt,
    OffsetDateTime createdAt) {}

File: apps/backend-node/src/features/notifications/dto/notification.dto.ts
import type { NotificationType } from './notification-type.dto';

export interface NotificationDto {
  id: string;
  userId: string;
  type: NotificationType;
  title: string;
  body: string;
  metadata: Record<string, unknown>;
  readAt: string | null;
  createdAt: string;
}

export interface UnreadNotificationCountDto {
  count: number;
}

export interface MarkAllNotificationsReadResultDto {
  updatedCount: number;
}
